### PR TITLE
Adds Colored Lights for Energy Swords

### PR DESCRIPTION
**Pull request by TalkingCactus**
:cl: TalkingCactus
add: Energy swords (and other energy melee weapons) now have a colored light effect when active.
/:cl:

Also:

- Generalized color code for energy melee weapons to make it easier to add new colored weapons in the future.

- Fixed a typo in the energy axe's description.

Dark room:
darkroom
Bright room:
brightroom

### CODE REVIEW

**Comment by PKPenguin321** [comment]
But rainbow?

File: code/game/objects/items/weapons/melee/energy.dm [hunk 1]
```diff
@@ -12,6 +14,23 @@
 	max_integrity = 200
 	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 100, acid = 30)
 	resistance_flags = FIRE_PROOF
+	var/brightness_on = 3
+
+/obj/item/weapon/melee/energy/New()
```
Comment (Cyberboss) [inline code/game/objects/items/weapons/melee/energy.dm]: INITIALIZE!!!

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by ChangelingRain** [comment]
The red and green are really oversaturated compared to the blue and purple

**Comment by Kromgar** [comment]
I like oversaturated

**Comment by TalkingCactus** [comment]
Rainbow eswords will change color randomly every time they're processed, didn't want to go overboard and break something with it. Also, need to handle dual sabers.

**Comment by TalkingCactus** [comment]
The colors are ripped directly from the blade colors. I can saturate the blue and purple a bit more if you want.

**Comment by GunHog** [comment]
Time for a murdering lights party!

File: code/game/objects/items/weapons/melee/energy.dm [hunk 2]
```diff
@@ -4,6 +4,8 @@
 	var/throwforce_on = 20
 	var/icon_state_on = "axe1"
 	var/list/attack_verb_on = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	var/random_color = 0
+	var/list/possible_colors = list()
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/melee/energy.dm]: null by default

File: code/game/objects/items/weapons/melee/energy.dm [hunk 3]
```diff
@@ -4,6 +4,8 @@
 	var/throwforce_on = 20
 	var/icon_state_on = "axe1"
 	var/list/attack_verb_on = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	var/random_color = 0
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/melee/energy.dm]: remove this, simply check if ```possible_colors``` exists and has elements, (```LAZYLEN(possible_colors)```)

File: code/game/objects/items/weapons/melee/energy.dm [hunk 4]
```diff
@@ -12,6 +14,23 @@
 	max_integrity = 200
 	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 100, acid = 30)
 	resistance_flags = FIRE_PROOF
+	var/brightness_on = 3
+
+/obj/item/weapon/melee/energy/Initialize()
+	..()
+	if(random_color && possible_colors.len)
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/melee/energy.dm]: just ```if(LAZYLEN(possible_colors))```

**Comment by jughu** [comment]
RAVE MURDERBONE!!!

**Comment by TalkingCactus** [comment]
bzzt

**Pull request merged**